system/Database/ConnectionInterface.php

```diff
@@ -193,4 +192,12 @@ public function escape($str);
 	public function callFunction(string $functionName, ...$params);
 
 	//--------------------------------------------------------------------
+
+	/**
+	 * Determines if the statement is a write-type query or not.
+	 *
+	 * @param  string $sql
+	 * @return boolean
+	 */
+	public function isWriteType($sql): bool;
```

Comment:
@michalsn :
I have changed the two tests you pointed out in an attempt to run the tests for `DBDebug=true` and `DBDebug=false`. As I mentioned in this [prior](https://github.com/codeigniter4/CodeIgniter4/pull/4176#discussion_r582159687) [comments](https://github.com/codeigniter4/CodeIgniter4/pull/4176#discussion_r582171805) I had some concern about what effect this might have on other unit tests. My solution assumes the tests are executed in the order they are defined in a given unit test file.

I have modified the changelog and upgrade documents as you requested. I hope that these changes satisfy your change request.

Reply:
@michalsn It looks good to me.
If you have no problems, merge this, please.